File: 2008/qualification/A.java
import java.util.*;
import java.io.*;

public class A {
    public static void main(String[] args) {
        Scanner in = new Scanner(new BufferedReader(new InputStreamReader(System.in)));

        int t = in.nextInt();
        for (int ti = 0; ti < t; ti++) {
            System.out.println("Case #" + (ti + 1) + ": " + new A().solveCase(in));
        }
    }

    private int solveCase(Scanner in){
        Set<String> engines = new HashSet<>();
        Set<String> queries = new HashSet<>();

        int s = in.nextInt();
        in.nextLine();

        for (int i = 0; i < s; i++) {
            String engine = in.nextLine();
            engines.add(engine);
        }

        int q = in.nextInt();
        in.nextLine();

        int count = 0;
        for (int j = 0; j < q; j++) {
            String query = in.nextLine();
            queries.add(query);

            if (queries.size() == engines.size()){
                count++;
                queries.clear();
                queries.add(query);
            }
        }

        return count;
    }
}

File: 2008/qualification/a.py
def solve():
    t = int(raw_input())
    for i in xrange(1, t + 1):
        result = solve_case()

        print 'Case #{}: {}'.format(i, result)


def solve_case():
    engines = set({})
    queries = set({})

    s = int(raw_input())
    for i in range(s):
        engines.add(raw_input())

    count = 0

    q = int(raw_input())
    for j in range(q):
        query = raw_input()
        queries.add(query)

        if len(engines) == len(queries):
            count += 1
            queries.clear()
            queries.add(query)

    return count


solve()
